integreat_compass/cms/views/offers/offer_form_view.py

```diff
@@ -0,0 +1,178 @@
+from django.db import transaction
+from django.shortcuts import render
+from django.utils.decorators import method_decorator
+from django.utils.translation import gettext as _
+from django.views.generic import TemplateView
+
+from ...decorators import permission_required
+from ...forms import (
+    ContactForm,
+    LocationForm,
+    OfferForm,
+    OfferVersionForm,
+    OrganizationForm,
+)
+from ...models import Offer, User
+
+logger = logging.getLogger(__name__)
+
+
+@method_decorator(permission_required("cms.edit_offer"), name="dispatch")
+class OfferFormView(TemplateView):
+    """
+    View for the offer form
+    """
+
+    template_name = "offers/offer_form.html"
+
+    def get(self, request, *args, **kwargs):
+        r"""
+        Render :class:`~integreat_compass.cms.forms.offers.contact_form.ContactForm`,
+               :class:`~integreat_compass.cms.forms.offers.location_form.LocationForm`,
+               :class:`~integreat_compass.cms.forms.offers.offer_form.OfferForm`,
+               :class:`~integreat_compass.cms.forms.offers.offer_version_form.OfferVersionForm` and
+               :class:`~integreat_compass.cms.forms.offers.organization_form.OrganizationForm`
+
+        :param request: The current request
+        :type request: ~django.http.HttpRequest
+
+        :param \*args: The supplied arguments
+        :type \*args: list
+
+        :param \**kwargs: The supplied keyword arguments
+        :type \**kwargs: dict
+
+        :raises ~django.core.exceptions.PermissionDenied: If user does not have the permission to edit the specific page
+
+        :return: The rendered template response
+        :rtype: ~django.template.response.TemplateResponse
+        """
+
+        offer = Offer.objects.filter(id=kwargs.get("offer_id")).first()
+        (offer_version, offer_contact, offer_location, offer_organization) = (
+            (
+                offer.versions.latest(),
+                offer.offer_contact,
+                offer.location,
+                offer.organization,
+            )
+            if offer is not None
+            else (None, None, None, None)
+        )
+        offer_form = OfferForm(instance=offer)
+        offer_version_form = OfferVersionForm(instance=offer_version)
+        contact_form = ContactForm(instance=offer_contact)
+        location_form = LocationForm(instance=offer_location)
+        organization_form = OrganizationForm(instance=offer_organization)
+
+        return render(
+            request,
+            self.template_name,
+            {
+                **self.get_context_data(**kwargs),
+                "offer_form": offer_form,
+                "offer_version_form": offer_version_form,
+                "contact_form": contact_form,
+                "location_form": location_form,
+                "organization_form": organization_form,
+            },
+        )
+
+    # pylint: disable=too-many-locals
+    @transaction.atomic
+    def post(self, request, *args, **kwargs):
```

Comment:
This function is to big, please split it up.

There are different task this functions fullfills:
1) Check permissions
2) Load User and append to different forms
3) Validate forms
4) Save forms

Reply:
The permission check and loading the user are gone now. I don't really see a reason to split up the remaining method - all we are doing is getting and saving the forms. It makes sense to do this in one method IMO.

It's just that we have so many forms that this produces numerous lines, but since this can't really be generalized into a (general) helper method, I think keeping it as it is now is the most readable version.